File: src/test/java/ukalus/persistence/Adder.java
package ukalus.persistence;

import java.io.Serializable;
import java.util.function.Function;

class Adder implements Function<Reference, Object>, Serializable {

  private static final long serialVersionUID = 3256999939061199155L;
  private long value;

  Adder(long value) {
    this.value = value;
  }

  /*
   * (non-Javadoc)
   * 
   * @see persistence.Command#execute(java.lang.Object)
   */
  public Object apply(Reference object) {
    return ((Counter) object.get()).add(value);
  }

}

File: src/test/kotlin/ukalus/persistence/Adder.kt
package ukalus.persistence

import java.io.Serializable
import java.util.function.Function

internal class Adder(private val value: Long) : Function<Reference<Counter?>, Any?>, Serializable {
    override fun apply(`object`: Reference<Counter?>): Any? {
        return (`object`.get() as Counter).add(value)
    }
}